compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2877,7 +2877,13 @@ public void visit(BLangSimpleVarRef varRefExpr, AnalyzerData data) {
                 actualType = varSym.type;
                 markAndRegisterClosureVariable(symbol, varRefExpr.pos, data.env, data);
             } else if ((symbol.tag & SymTag.TYPE_DEF) == SymTag.TYPE_DEF) {
-                actualType = symbol.type.tag == TypeTags.TYPEDESC ? symbol.type : new BTypedescType(symbol.type, null);
+                if (symbol instanceof BTypeDefinitionSymbol) {
+                    BTypeDefinitionSymbol typeDefSym = (BTypeDefinitionSymbol) symbol;
+                    actualType = symbol.type.tag == TypeTags.TYPEDESC ? typeDefSym.referenceType
+                            : new BTypedescType(typeDefSym.referenceType, null);
```

Comment:
What is the reference type is another reference type that is a typedesc?

Reply:
It will not a problem ryt?
We need the name of the type definition.

ex:- 

```
type Bar2 Bar;
type Bar int|float;

``` 

For the type definition `Bar2` we need the Bar2 type here ryt?